src/components/chips/js/chipsController.js

```diff
@@ -577,23 +577,34 @@ MdChipsCtrl.prototype.hasMaxChipsReached = function() {
 
 /**
  * Updates the validity properties for the ngModel.
+ *
+ * TODO add the md-max-chips validator to this.ngModelCtrl.validators so that the validation will
+ * be performed automatically.
  */
 MdChipsCtrl.prototype.validateModel = function() {
   this.ngModelCtrl.$setValidity('md-max-chips', !this.hasMaxChipsReached());
   this.ngModelCtrl.$validate(); // rerun any registered validators
 };
 
-MdChipsCtrl.prototype.updateNgModel = function() {
-  this.ngModelCtrl.$setViewValue(this.items.slice());
-  // TODO add the md-max-chips validator to this.ngModelCtrl.validators so that
-  // the validation will be performed automatically on $viewValue change
-  this.validateModel();
+/**
+ * Function to handle updating the model, validation, and change notification when a chip
+ * is added, removed, or changed.
+ * @param {boolean=} skipValidation true to skip calling validateModel()
+ */
+MdChipsCtrl.prototype.updateNgModel = function(skipValidation) {
+  this.ngModelCtrl.$setDirty();
+  if (!skipValidation) {
+    this.validateModel();
+  }
+  if (this.ngChange) {
+    this.ngChange(this.items);
```

Comment:
AFAICT, this will not be called correctly. Since you are defining a scope binding (via `ngChange: '&'`), you need to pass an object that maps arguments to values. For example, if you have `ng-change="onChange(items)"`, then you need to call it with `this.ngChange({items: this.items})`.

Reply:
The key has to be the same as what they used as variable name. Essentially, a `&` binding allows you to evaluate an expression on the parent scope (of an isolate scope) and pass certain data. For security reasons, AngularJS creates and passes a wrapper to the isolate scope. So, the isolate scope does not get direct access to the expression (most often a method). What it gets is a wrapper function that takes a map of argument names to values and calls the expression on the parent scope with these values.

For example, having a `parentMethod: '&'` binding and using the component as `parent-method="doStuff(arg1, arg2)"`, allows the component's isolate scope to call `this.parentMethod({arg1: 'value1', arg2: 'value2'})`, which will eventually call `doStuff('value1', 'value2')` on the parent scope.

So, you need to know what arguments are used in the expression.

##
Usually, we don't have this issue with `ngChange`, because there is no isolate scope. In cases like this (where you have an isolate scope and you can't just `.$eval(attrs.ngChange)`), a common approach is to specify (i.e. document) that in the `ngChange` expression the chips will be available as `$chips` or `$items` (or whatever name you choose).

Then, people can use `ng-change="onChipsChange($chips)"` and you can call it with `this.ngChange({$chips: this.items})` (and their `onChipsChange()` method will be called with `this.items` as argument).

Reply:
Thank you very much for the explanation! That is quite helpful.

It looks like this is not needed due to the suggested change in https://github.com/angular/material/pull/11310#discussion_r192338790.